hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -173,6 +174,40 @@ AcT toCellularAccessTechnology(int rat) {
 // see 3GPP TS 45.008 [20] subclause 8.2.4
 const char compatQualMap[] = { 49, 43, 37, 25, 19, 13, 7, 0 };
 
+Dev fromPlatformNcpIdentifer(PlatformNCPIdentifier identifer) {
```

Comment:
This kind of duplicates some of the code from `cellular_device_info()` in `/hal/network/ncp/cellular/cellular_hal.cpp` but I struggled to find an appropriate place to make a helper function to map `PlatformNCPIdentifier` to `Dev` enum values. I settled on this approach since it seems `mdm_hal` looks to be the only file that uses the `Dev` enum extensively.

Reply:
Or `cellular_internal.h/cpp` might also be another place for this and the implementation there can be used in both `mdm_hal` and `cellular_device_info`.

Reply:
I was trying to find a shared spot that could be included from gen 2 `mdm_hal.cpp` as well as gen3 `cellular_hal.cpp`. `cellular_internal.cpp` seems to be only used by electron? Should I try to include this for gen3 builds?

Reply:
In that case `hal/shared` would be your best candidate. Similar to how `cellular_sig_perc_mapping`-stuff is implemented there and shared across Gen 2 and Gen 3.